src/Features/Core/Portable/ConvertTupleToStruct/AbstractConvertTupleToStructCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,865 @@
+            //
+            // Direct Dependencies:
+            //  Pros:
+            //      All updated projects are able to see the newly added type.
+            //      Transitive deps won't be updated to use a type they can't actually use.
+            //  Cons:
+            //      If that project then exports that new type, then transitive deps will
+            //      break if they use those exported APIs since they won't know about the
+            //      type.
+            //
+            // Transitive Dependencies:
+            //  Pros:
+            //      All affected code is updated.
+            //  Cons: 
+            //      Non-direct deps will not compile unless the take a reference on the
+            //      starting project.
+
+            var dependentProjects = graph.GetProjectsThatDirectlyDependOnThisProject(startingProject.Id);
+            var allProjects = dependentProjects.Select(solution.GetProject).Concat(startingProject).ToSet();
+
+            var result = ArrayBuilder<DocumentToUpdate>.GetInstance();
+            var tupleFieldNames = tupleType.TupleElements.SelectAsArray(f => f.Name);
+
+            foreach (var project in allProjects)
+            {
+                await AddDocumentsToUpdateForProjectAsync(
+                    project, result, tupleFieldNames, cancellationToken).ConfigureAwait(false);
+            }
+
+            return result.ToImmutableAndFree();
+        }
+
+        private async Task<ImmutableArray<DocumentToUpdate>> GetDocumentsToUpdateForContainingProjectAsync(
+            Project project, INamedTypeSymbol tupleType, CancellationToken cancellationToken)
+        {
+            var result = ArrayBuilder<DocumentToUpdate>.GetInstance();
+            var tupleFieldNames = tupleType.TupleElements.SelectAsArray(f => f.Name);
+
+            await AddDocumentsToUpdateForProjectAsync(
+                project, result, tupleFieldNames, cancellationToken).ConfigureAwait(false);
+
+            return result.ToImmutableAndFree();
+        }
+
+        private async Task AddDocumentsToUpdateForProjectAsync(Project project, ArrayBuilder<DocumentToUpdate> result, ImmutableArray<string> tupleFieldNames, CancellationToken cancellationToken)
+        {
+            foreach (var document in project.Documents)
+            {
+                var info = await document.GetSyntaxTreeIndexAsync(cancellationToken).ConfigureAwait(false);
+                if (info.ContainsTupleExpressionOrTupleType &&
+                    InfoProbablyContainsTupleFieldNames(info, tupleFieldNames))
+                {
+                    // Use 'default' for nodesToUpdate so we walk the entire document
+                    result.Add(new DocumentToUpdate(document, nodesToUpdate: default));
+                }
+            }
+        }
+
+        private bool InfoProbablyContainsTupleFieldNames(SyntaxTreeIndex info, ImmutableArray<string> tupleFieldNames)
```

Comment:
>InfoProbablyContainsTupleFieldNames [](start = 21, length = 35)

static?
Then maybe callers might be made static too?

Reply:
Done.